rrule/rrule.d.ts

```diff
@@ -0,0 +1,249 @@
+        count?: number;
+        until?: Date;
+        bysetpos?: number;
+        bymonth?: number;
+        bymonthday?: number;
+        byyearday?: number;
+        byweekno?: number;
+        byweekday?: Weekday[];
+        byhour?: number;
+        byminute?: number;
+        bysecond?: number;
+    }
+
+    interface Weekday {
+        new (n: number): Weekday;
+        nth(n: number): Weekday;
+        equals(other: Weekday): boolean;
+        toString(): string;
+        getJsWeekday(): number;
+    }
+
+}
+
+declare class RRule {
+
+    /**
+    * @param {Object?} options - see <http://labix.org/python-dateutil/#head-cf004ee9a75592797e076752b2a889c10f445418>
+    *        The only required option is `freq`, one of RRule.YEARLY, RRule.MONTHLY, ...
+    * @constructor
+    */
+    constructor(options: RRule.Options, noCache?: boolean);
+
+    options: RRule.Options;
+
+    origOptions: RRule.Options;
+
+    /**
+    * Returns the first recurrence after the given datetime instance.
+    * The inc keyword defines what happens if dt is an occurrence.
+    * With inc == True, if dt itself is an occurrence, it will be returned.
+    * @return Date or null
+    */
+    after(dt: Date, inc?: boolean): Date;
+
+    /**
+    * @param {Function} iterator - optional function that will be called
+    *                   on each date that is added. It can return false
+    *                   to stop the iteration.
+    * @return Array containing all recurrences.
+    */
+    all(iterator?: (date: Date) => void): Date[];
+
+    /**
+    * Returns all the occurrences of the rrule between after and before.
+    * The inc keyword defines what happens if after and/or before are
+    * themselves occurrences. With inc == True, they will be included in the
+    * list, if they are found in the recurrence set.
+    * @return Array
+    */
+    between(a: Date, b: Date, iterator?: (date: Date) => void): Date[];
+
+    /**
+    * Returns the last recurrence before the given datetime instance.
+    * The inc keyword defines what happens if dt is an occurrence.
+    * With inc == True, if dt itself is an occurrence, it will be returned.
+    * @return Date or null
+    */
+    before(dt: Date, inc?: boolean): Date;
+
+    /**
+    * Returns the number of recurrences in this set. It will have go trough
+    * the whole recurrence, if this hasn't been done before.
+    */
+    count(): number;
+
+    /**
+    * Converts the rrule into its string representation
+    * @see <http://www.ietf.org/rfc/rfc2445.txt>
+    * @return String
+    */
+    toString(): string;
+
+    /**
+    * Will convert all rules described in nlp:ToText
+    * to text.
+    */
+    toText(gettext?: (str: string) => string, language?: any): string;
+
+    isFullyConvertibleToText(): boolean;
+
+    clone(): RRule;
+
+}
+
+declare namespace RRule {
+
+    const FREQUENCIES: ['YEARLY' | 'MONTHLY' | 'WEEKLY' | 'DAILY' | 'HOURLY' | 'MINUTELY' | 'SECONDLY'];
+
+    const YEARLY: number;
```

Comment:
The number values of these don't mean anything, so these should be typed as an enum. This may be difficult since you want `RRule.MONTLY` and not `RRule.Period.MONTHLY`, but you could define a `enum Period {}` and have `const MONTHLY: Period;` and so on.